internal/app/service_test.go

```diff
@@ -604,56 +605,56 @@ func TestLoadConfigurableFunctionPipelinesNumFunctions(t *testing.T) {
 	assert.Equal(t, expectedTransformsCount, len(pipeline.Transforms))
 }
 
-func TestUseTargetTypeOfByteArrayTrue(t *testing.T) {
+func TestTargetType(t *testing.T) {
 	functions := make(map[string]common.PipelineFunction)
 	functions["Compress"] = common.PipelineFunction{
 		Parameters: map[string]string{Algorithm: CompressGZIP},
 	}
 	functions["SetResponseData"] = common.PipelineFunction{}
-
-	sdk := Service{
-		lc: lc,
-		config: &common.ConfigurationStruct{
-			Writable: common.WritableInfo{
-				Pipeline: common.PipelineInfo{
-					ExecutionOrder:           "Compress, SetResponseData",
-					UseTargetTypeOfByteArray: true,
-					Functions:                functions,
-				},
-			},
-		},
+	tests := []struct {
+		name               string
+		targetTypeName     string
+		expectedTargetType interface{}
+		expectErr          bool
+	}{
+		{"raw", rawTargetType, &[]byte{}, false},
+		{"metric", metricTargetType, &dtos.Metric{}, false},
+		{"event", eventTargetType, &dtos.Event{}, false},
+		{"empty", emptyTargetType, &dtos.Event{}, false},
+		{"raW wrong case", "raW", nil, true},
+		{"eveNt wrong case", "eveNt", nil, true},
+		{"meTric wrong case", "meTric", nil, true},
```

Comment:
These should work. i.e. same as lower case once you address my other comment in the code.

Reply:
I wasnt sure if it should work or not. thank you for the clarification. I have made the changes